src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +32,13 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        Pattern p = Pattern.compile(ADDRESS_SPLIT_REGEX);
+        String[] items = p.split(address);
+        block = new Block(items[0]);
```

Comment:
What is one component is missing. can try to handle it rather than forcing 4 components.